Integrations/integration-Thinkst_Canary.yml

```diff
@@ -0,0 +1,558 @@
+            token_file = fileResult(name, content)
+            demisto.results(token_file)
+        elif res.get('token').get('web_image'):
+            name = res.get('token').get('web_image_name')
+            content = res.get('token').get('web_image')
+            token_file = fileResult(name, content)
+            demisto.results(token_file)
+        else:
+            results['HumanReadable'] = tableToMarkdown('Canary Tools Tokens', res.get('token'))
+        demisto.results(results)
+
+
+    def check_whitelist(ip, port):
+        # Check if a given IP address is whitelisted in Canary Tools
+        params = {
+            'auth_token': AUTH_TOKEN,
+            'src_ip': ip,
+            'dst_port': port
+        }
+
+        res = session.request('GET', SERVER + 'settings/is_ip_whitelisted', params=params, verify=not UNSECURED).json()
+        return res
+
+
+    def check_whitelist_command():
+        ip = demisto.args().get('ip')
+        port = demisto.args().get('port')
+        res = check_whitelist(ip, port)
+
+        if not port:
+            port = 'Any'
+        context = {
+            'Address': '{}'.format(ip),
+            'Port': '{}'.format(port),
+            'Whitelisted': '{}'.format(res.get('is_ip_whitelisted'))
+        }
+        if res.get('is_ip_whitelisted'):
+            demisto.results({
+                'Type': entryTypes['note'],
+                'ContentsFormat': formats['json'],
+                'Contents': res,
+                'ReadableContentsFormat': formats['markdown'],
+                'HumanReadable': 'The IP address ' + '{}'.format(ip) + ':' + '{}'.format(port) + ' is Whitelisted!',
+                'EntryContext': {
+                    'CanaryTools.IP(val.Address && val.Address == obj.Address && val.Port && val.Port == obj.Port)': context}
+            })
+        else:
+            demisto.results({
+                'Type': entryTypes['note'],
+                'ContentsFormat': formats['json'],
+                'Contents': res,
+                'ReadableContentsFormat': formats['markdown'],
+                'HumanReadable': 'The IP address ' + '{}'.format(ip) + ':' + '{}'.format(port) + ' is not Whitelisted!',
+                'EntryContext': {
+                    'CanaryTools.IP(val.Address && val.Address == obj.Address && val.Port && val.Port == obj.Port)': context}
+            })
+
+
+    def whitelist_ip(ip, port):
+        # Whitelist an IP address in Canary Tools
+        params = {
+            'auth_token': AUTH_TOKEN,
+            'src_ip': ip,
+            'dst_port': port
+        }
+
+        res = session.request('POST', SERVER + 'settings/whitelist_ip_port', params=params, verify=not UNSECURED).json()
+        return res
+
+
+    def whitelist_ip_command():
+        ip = demisto.args().get('ip')
+        port = demisto.args().get('port')
+        res = whitelist_ip(ip, port)
+
+        if not port:
+            port = 'Any'
+        if res.get('result') == 'success':
```

Comment:
first initialize `res.get('result')` in an argument and then use it in the conditions